src/tools/crawler.rs

```diff
@@ -369,13 +369,26 @@ mod tests {
 
         sleep(Duration::from_secs(1)).await;
 
+        // Capture the start time of the crawler.
+        let crawler_start_time = Instant::now();
+
         tokio::spawn(async move {
             loop {
                 crawler.known_network.update_nodes();
 
                 info!(parent: crawler.node().span(), "asking peers for their peers (connected to {})", crawler.node().num_connected());
                 info!(parent: crawler.node().span(), "known addrs: {}", crawler.known_network.num_nodes());
 
+                let good_nodes: HashMap<SocketAddr, KnownNode> = crawler.known_network.nodes()
```

Comment:
```suggestion
                let good_nodes = crawler.known_network.nodes()
```
no benefit from using a `HashMap` or collecting here